scipy/stats/_stats_py.py

```diff
@@ -486,6 +497,11 @@ def mode(a, axis=0, nan_policy='propagate', keepdims=False):
     count : ndarray
         Array of counts for each mode.
 
+    Raises
+    ------
+    TypeError
+        If `a` is not numeric.
```

Comment:
This one is not desirable. It's too generic; `a` is very clearly documented as a numerical array already.